### DISCUSSION TITLE
How do I change providers?

### DISCUSSION THREAD

**Discussion by PrivatePolonsky** [pystardust/ani-cli#1010] [Q&A] [answered]
Used to be able to choose what website (Provider) to scrape from  with  "-r" option,
but it is now used for episode selection. how would I be able to scrape a different provider without editing the script?

**Comment by port19x** [comment, marked as answer]
This is not supported anymore.
If we reintroduce the functionality in the future it might be an environment variable.
Why would you want that anyway? 
If you're concerned with coverage, providers have failover for anime that are available at provider A, but not provider B.
For quality, `-q` should be sufficient.
If your firewall blocks one or more providers, failover should once again handle that

**Reply by PrivatePolonsky** [reply]
trying to watch via allanime the episode i am watching is only 1 minute long
using gogoanime it would be full length (about 25 min)

https://allanime.site/watch/aqFgGPzgLqDFQLnvr/toradora-/episode-20-dub

https://www1.gogoanime.bid/toradora-dub-episode-20

**Reply by ghost** [reply]
gogoanime.bid is a fake website though. Go to https://gogoanime.news so see the real domains.

**Reply by PrivatePolonsky** [reply]
> gogoanime.bid is a fake website though. Go to https://gogoanime.news so see the real domains.

That's the website it redirects me to though.

**Reply by CoolnsX** [reply]
why don't you watch directly from source, where gogonanime takes urls from? it's their Inhouse service, but many people doesn't seem to know about it, and also it's search engine is trash, but if you want latest anime with minimum ads ( only show ads in player) then here is the site https://anihdplay.com/

**Comment by ghost** [comment]
Having the same problem here. I want to watch `Koe no katachi (silent voice)` in dub but whenever I try to search for the anime using the `--dub` parameter. It shows the name of the anime but when I click on it, it doesn't launch mpv or vlc. Try it for yourself if you don't believe me. `Koe no katachi (silent voice)` dub version is already there in gogoanime.

EDIT : Tbh allanime is a shitty website.

**Reply by port19x** [reply]
can you open an issue for that?

**Reply by ghost** [reply]
> can you open an issue for that?

https://github.com/pystardust/ani-cli/issues/1017
